### PR TITLE
React cart management

### PR DESCRIPTION
**Pull request by gopalbbharadva**
PR For Cart Management.

### Features:
- Add/remove  to/from  cart respectively.
- Bill calculation.
- Increment decrement item quantity.
- Toggle button add to cart TO go to cart on product listing page.

P.S: Check only last 3 commits, as it's only for cart management. It shows all the commits till I stop merge. I Will merge soon after some changes.

Live link:https://competent-darwin-57aa93.netlify.app/

### CODE REVIEW

**Review by tanveertkd — APPROVED** [review]
This looks good. 👍

File: src/pages/Wishlist/Wishlist.jsx [hunk 1]
```diff
@@ -1,12 +1,32 @@
...
+import { useWishList } from "../../contexts/WishListContext";
+import { WishListService } from "../../Services/WishListService";
 import "./Wishlist.css";
 
 export const Wishlist = () => {
+  const { currentUser } = useAuth();
+  const { wishList, userWishList } = useWishList();
+
+  WishListService();
+  // console.log(wishList.length);
```
Comment (msdeshmukh009) [inline src/pages/Wishlist/Wishlist.jsx]: Unwanted comment I guess.

File: src/Services/CartService.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  const navigate = useNavigate();
+  const { cartDispatch } = useCart();
+
+  useEffect(() => {
+    (async () => {
+      if (!token) {
+        navigate("/login");
+        return;
+      } else {
+        console.log("else part");
```
Comment (msdeshmukh009) [inline src/Services/CartService.js]: You forgot to remove this console I guess.

File: src/reducers/CartReducer.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  cartStatus: 0,
+  bill: 0,
+  discount: 0,
+};
+
+const cartReducer = (state, action) => {
+  switch (action.type) {
+    case "SET_CART_FROM_API":
+      return { ...state, cartArray: [...action.payload] };
+    case "ADD_TO_CART":
```
Comment (msdeshmukh009) [inline src/reducers/CartReducer.js:17]: You can use constants for action types.

**Review by msdeshmukh009 — APPROVED** [review]
The code looks nice. I Liked how you have implemented the custom hooks. 
**A UX-related suggestion:** 
- Have a link to the product listing page in your navbar. 
- Also, there is a slight horizontal overflow on the product listing page. 
- It will be good if users can toggle wishlist items from the product listing page.

**Pull request merged**